### PR TITLE
Add `validate_tags` util to check that tags are an array of strings

### LINKED ISSUE #50: Ensure that the "tags" parameter is always a list
When the user sets the `tags` option with something, ensure that it is just a list of strings (not something nested), it will warn the user that the point tag set is malformed and avoid rejecting it much later in the backend.

### PR DESCRIPTION
**Pull request by zippolyte**
### What does this PR do?

SSIA
fixes #50 

### Verification Process

Unit tests

### Review checklist (to be filled by reviewers)

- [ ] Feature or bug fix MUST have appropriate tests (unit, integration, etc...)
- [ ] PR title must be written as a CHANGELOG entry [(see why)](/CONTRIBUTING.md#pull-request-title)
- [ ] Files changes must correspond to the primary purpose of the PR as described in the title (small unrelated changes should have their own PR)
- [ ] PR must have one `changelog/` label attached. If applicable it should have the `backward-incompatible` label attached.
- [ ] PR should not have `do-not-merge/` label attached.
- [ ] If Applicable, issue must have `kind/` and `severity/` labels attached at least.

### CODE REVIEW

File: lib/dogapi/common.rb [hunk 1]
```diff
@@ -205,4 +205,13 @@ def Dogapi.find_localhost
     raise $ERROR_INFO unless $ERROR_INFO.class.name == 'Errno::ENOENT'
     @@hostname = Addrinfo.getaddrinfo(Socket.gethostname, nil, nil, nil, nil, Socket::AI_CANONNAME).first.canonname
   end
+
+  def Dogapi.validate_tags(tags)
+    unless tags.is_a? Array
```
Comment (gzussa) [inline lib/dogapi/common.rb]: I am not this kind of precious stone expert but what if tags is null?

**Comment by miketheman** [comment]
What calls validate_tags??

**Comment by zippolyte** [comment]
It will be up to the user to call that utility function in order to validate the tags they pass to the API.

**Comment by miketheman** [comment]
That seems counter to the original report.
There doesn't seem to be a change log or docs update with this, so how would a user know? Why make it the user responsibility, when you can call it for them?

**Comment by zippolyte** [comment]
True this doesn't exactly answer the original report, I'll leave a comment on the issue explaining our reasoning with this.

We didn't want to add this validation to every endpoint that manipulates tags because it can cause some performance drop given that we have to check for every individual tag.
Additionally, given this report never gained much traction, we didn't want to make such a change to every single endpoint and risk breaking things for current users.

However, if there is a real need for this, we now at least provide this utility function for a user to easily validate their tags, an for us to later add this to every endpoint.
A changelog entry will be added when the release is made for users to know about this, plus a comment on the original issue.

**Pull request merged**